src/v/cluster/topics_frontend.cc

```diff
@@ -521,25 +526,30 @@ ss::future<topic_result> topics_frontend::replicate_create_topic(
     }
 
     return replicate_and_wait(_stm, _as, std::move(cmd), timeout)
-      .then_wrapped([tp_ns = std::move(tp_ns), units = std::move(units)](
-                      ss::future<std::error_code> f) mutable {
-          try {
-              auto error_code = f.get0();
-              auto ret_f = ss::now();
-              return ret_f.then(
-                [tp_ns = std::move(tp_ns), error_code]() mutable {
-                    return topic_result(std::move(tp_ns), map_errc(error_code));
-                });
+      .then_wrapped(
+        [tp_ns = std::move(tp_ns),
+         units = std::move(units),
+         &tt = _topics.local()](ss::future<std::error_code> f) mutable {
```

Comment:
Oh yeah, this seems really iffy. I assume we don't want to access the topic table until **after** we've replicated the create command, but this usage _feels_ unsafe.